dbms/src/IO/ReadHelpers.cpp

```diff
@@ -177,7 +189,10 @@ void readStringInto(Vector & s, ReadBuffer & buf)
     {
         char * next_pos = find_first_symbols<'\t', '\n'>(buf.position(), buf.buffer().end());
 
-        appendToStringOrVector(s, buf.position(), next_pos);
+        if (buf.isPadded())
```

Comment:
Maybe we can put it inside one another inline function?

Reply:
Yeah, there is indeed a chance.

Reply:
btw, recent revisions to FunctionComparisor make g++ 8.2 mad. It takes 4 minutes to compile the translation unit.

Reply:
Yes, something is over-complicated in implementation of DECIMAL data type. We will address it.